lib/ansible/modules/network/avi/avi_actiongroupconfig.py

```diff
@@ -0,0 +1,201 @@
+
+short_description: ActionGroupConfig Configuration
+description:
+    - This module is used to configure ActionGroupConfig object
+    - more examples at <https://github.com/avinetworks/avi-ansible-samples>
+requirements: [ avisdk ]
+version_added: 2.3
+options:
+    controller:
+        description:
+            - location of the controller. Environment variable AVI_CONTROLLER is default
+    username:
+        description:
+            - username to access the Avi. Environment variable AVI_USERNAME is default
+    password:
+        description:
+            - password of the Avi user. Environment variable AVI_PASSWORD is default
+    tenant:
+        description:
+            - tenant for the operations
+        default: admin
+    tenant_uuid:
+        description:
+            - tenant uuid for the operations
+        default: ''
+    state:
+        description:
+            - The state that should be applied on the entity.
+        required: false
+        default: present
+        choices: ["absent","present"]
+    action_script_config_ref:
+        description:
+            - reference of the action script configuration to be used object ref AlertScriptConfig.
+        type: string
+    autoscale_trigger_notification:
+        description:
+            - Trigger Notification to AutoScale Manager
+        default: False
+        type: bool
+    description:
+        description:
+            - Not present.
+        type: string
+    email_config_ref:
+        description:
+            - Select the Email Notification configuration to use when sending alerts via email. object ref AlertEmailConfig.
+        type: string
+    external_only:
+        description:
+            - Generate Alert only to external destinations
+        required: true
+        type: bool
+    level:
+        description:
+            - When an alert is generated, mark its priority via the Alert Level.
+        required: true
+        type: string
+    name:
+        description:
+            - Not present.
+        required: true
+        type: string
+    snmp_trap_profile_ref:
+        description:
+            - Select the SNMP Trap Notification to use when sending alerts via SNMP Trap. object ref SnmpTrapProfile.
+        type: string
+    syslog_config_ref:
+        description:
+            - Select the Syslog Notification configuration to use when sending alerts via Syslog. object ref AlertSyslogConfig.
+        type: string
+    tenant_ref:
+        description:
+            - Not present. object ref Tenant.
+        type: string
+    url:
+        description:
+            - url
+        required: true
+        type: string
+    uuid:
+        description:
+            - Not present.
+        type: string
+'''
+
+RETURN = '''
+obj:
+    description: ActionGroupConfig (api/actiongroupconfig) object
+    returned: success, changed
+    type: dict
+'''
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
```

Comment:
1) Common arguments (which you should document in a `docs_fragment` should be pulled out of here, see
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/f5/bigip_device_dns.py#L374

And corresponding code in https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/f5.py#L39

2) Please reformat so each option is on a singleline, and fix indentation.

3) Secret items, such as `password` should have ` no_log=True`